app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoMegaTile.kt

```diff
@@ -54,7 +29,31 @@ class PocketVideoMegaTile(
         gravity = Gravity.CENTER_VERTICAL or Gravity.END
 
         LayoutInflater.from(context).inflate(R.layout.pocket_video_mega_tile, this, true)
-        pocketWordmarkView.setImageDrawableAsPocketWordmark()
-        thumbnailViews = listOf(thumbnail1View, thumbnail2View, thumbnail3View, thumbnail4View)
+        PocketDrawable.setImageDrawableAsPocketWordmark(pocketWordmarkView)
+    }
+
+    fun setContent(feedItems: List<PocketFeedItem>) {
+        listOf(thumbnail1View, thumbnail2View, thumbnail3View, thumbnail4View)
+            .zip(feedItems)
+            .forEach { (view, item) ->
+                when (item) {
+                    is PocketFeedItem.Video -> showFetchedImage(view, item.thumbnailURL)
+                    is PocketFeedItem.Loading -> showLocalDrawable(view, item.thumbnailResource)
+                }
+            }
+    }
+
+    private fun showLocalDrawable(view: ImageView, drawable: Int) {
+        view.setImageResource(drawable)
+    }
+
+    private fun showFetchedImage(imageView: ImageView, url: String) {
+        val roundCornerTransformation =
+            RoundCornerTransformation(resources.getDimension(R.dimen.pocket_video_mega_tile_thumbnail_corner_radius))
+
+        PicassoWrapper.client.load(url)
```

Comment:
nit: maybe we can provide better UX by using `onImageLoadFailed`

Reply:
Hm, I don't see the option.  Does `placeholder` take care of that?